services/agent-router/internal/httpapi/place_test.go

```
@@ -0,0 +1,580 @@
+	t.Run("edge_only", func(t *testing.T) {
+		e := newEnv(t, realCatalogState(t))
+		e.do(t, http.MethodPost, "/v1/capacity/heartbeat", nodeToken,
+			placeHeartbeatBody(nodeName, "INTERACTIVE", strp("qwen36-27b"), []string{"qwen36-27b"}, 24))
+
+		raw := readAll(t, postPlace(t, e, callerToken, placeRequestBody(t, localCodeStandardBody("edge-only"))))
+		testutil.ValidateJSON(t, schema, raw)
+		p := decodeRaw[placeResult](t, raw)
+		if p.Status != "unavailable" || p.Placement != nil {
+			t.Fatalf("result = %+v, want an explicit unavailable result, never a substitute placement", p)
+		}
+		if p.Headers != nil && len(p.Headers) != 0 {
```

golangci-lint (2.12.2)

[error] S1009: should omit nil check; len() for nil maps is defined as zero

(staticcheck)